src/sema/types.rs

```diff
@@ -399,30 +378,102 @@ fn resolve_contract<'a>(
                 delay.events.push(ResolveEventFields {
                     event_no,
                     pt,
-                    comments: tags,
+                    comments: parts.doccomments.clone(),
                     contract: Some(contract_no),
                 });
             }
             pt::ContractPart::TypeDefinition(ty) => {
-                let tags = parse_doccomments(comments, doc_comment_start, ty.loc.start());
+                annotions_not_allowed(&parts.annotations, "type", ns);
 
-                type_decl(ty, file_no, &tags, Some(contract_no), ns);
-            }
-            pt::ContractPart::FunctionDefinition(f) => {
-                if let Some(pt::Statement::Block { loc, .. }) = &f.body {
-                    doc_comment_start = loc.end();
-                    continue;
-                }
+                type_decl(ty, file_no, &parts.doccomments, Some(contract_no), ns);
             }
             _ => (),
         }
-
-        doc_comment_start = parts.loc().end();
     }
 
     broken
 }
 
+fn contract_annotations(
```

Comment:
Please, add a doc comment to this function.

Reply:
I don't know what to write in a doc comment. So, I've written:

```
/// Resolve annotations attached to a contract
```